packages/adp-tooling/src/types.ts

```diff
@@ -345,9 +334,32 @@ export interface AdpProjectData {
     id: string;
 }
 
+export interface ChangeDataSourceAnswers {
+    /** Data Source identifier  */
+    id: string;
+    /** Data Source URI */
+    uri: string;
+    /** Data Source Max Age */
+    maxAge?: number;
+    /** Data Source Annotation URI */
+    annotationUri?: string;
+}
+
+export type DataSource = ManifestNamespace.DataSource & { dataSourceName: string; annotations: string[] };
+
 export interface AdpCustomConfig {
     adp: {
         safeMode: boolean;
         environment: OperationsType;
     };
-}
\ No newline at end of file
+}
+
+export interface PreviewConfigAdp {
+    adp?: {
```

Comment:
That type is already defined at https://github.com/SAP/open-ux-tools/blob/main/packages/adp-tooling/src/types.ts#L20.

Btw, it also contains a flag for ignoring cert errors - maybe you could also read that because if it is set to true here (by the user) you can use it and don't have to prompt the users.